tests/Opus/CollectionRoleTest.php

```diff
@@ -1116,4 +1116,42 @@ public function testChangeOfHideEmptyCollectionsDoesNotAffectDocuments()
         $this->assertEquals($doc->getServerDateModified(), $serverDateModified);
     }
 
+    /**
+     * Validierung von gültigen CollectionRole Namen (OPUSVIER-4022)
+     */
+    public function testValidCollectionRoleName()
+    {
+        $collRole = new Opus_CollectionRole();
+        $collRole->setName('abcABC012_3-4_5-6');
+        $collRole->setOaiName('foo');
+        $collRoleId = $collRole->store();
+
+        $this->assertTrue($collRoleId > 0);
+    }
+
+    /**
+     * Validierung von ungültigen CollectionRole Namen (OPUSVIER-4022)
+     */
+    public function testInvalidCollectionRoleName()
+    {
+        $invalidNames = ['a b ', 'a,b', 'ä', 'a:b', 'a;b'];
+
+        foreach ($invalidNames as $invalidName) {
+            $collRole = new Opus_CollectionRole();
+            $collRole->setName($invalidName);
+            $collRole->setOaiName('foo');
+
+            $exception = null;
+            try {
+                $collRole->store();
+            }
+            catch (Opus_Model_Exception $exception) {
+                // do nothing
+            }
+
+            // setExpectedException kann bei Schleife nicht angewendet werden; beendet Test nach erster Exception
```

Comment:
Das verstehe ich noch nicht. Warum kann nicht `@expectedException` verwendet werden? Eine weitere Möglichkeit wäre `@dataProvider` und eine Test-Funktion, die ohne Schleife auskommt. Zu den invaliden Namen würde auch Varianten mit Zahlen, '-' oder '_' am Anfang gehören.

Reply:
Wenn `setExpectedException`gesetzt ist, dann wird die Schleife beim Auftreten der ersten Exception verlassen. Folglich wird nur das erste Muster `a b` geprüft.

Die invaliden Namensmuster erweitere ich.

Reply:
kurze Frage: welche Muster willst Du abgeprüft haben? Führende Ziffern, Unterstrich und Bindestrich sind nicht verboten.

Reply:
Da habe ich die Regular-Expression nur flüchtig gelesen. Verbiete mal bitte alles außer Buchstaben am Anfang. Das entscheide ich jetzt einfach, unter anderem weil zum Beispiel IDs in HTML nur mit Buchstaben anfangen dürfen. Es ist keine technische Anforderung für CSS-Klassen, so viel ich weiß, aber ich will damit erst gar nicht anfangen.